### PR TITLE
OpenJazz: Add a new recipe

### PR DESCRIPTION
**Pull request by przemub**
It is a recipe for a free re-implementation of the engine of Jazz Jackrabit - a platform game.
It's my first recipe, so I'd be thankful if someone took a careful look on it.

### CODE REVIEW

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 1]
```diff
@@ -0,0 +1,50 @@
+SUMMARY="A free, open-source version of classic Jazz Jackrabbit games"
+DESCRIPTION="A free, open-source version of classic Jazz Jackrabbit games.\
+Requirest the original game data to work. When launching OpenJazz, pass the location of the data, eg:\
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Should less Than 80 char

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+OpenJazz ~/jazz1
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Would better to use ~/config/non-packaged/Data/openjazz

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+"
+HOMEPAGE="https://github.com/AlisterT/openjazz"
+COPYRIGHT="Alister Thomson (alister_j_t at yahoo dot com)"
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Remove the Email address

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+COMMIT="0f59db3a6b0e2c076b39917dc65cfb3562611ee6"
+SOURCE_URI="https://github.com/AlisterT/openjazz/archive/$COMMIT.tar.gz"
+CHECKSUM_SHA256="55c82328ef3f0185951bd339be55411410d1fa786f391091e15b7a093a08fafd"
+SOURCE_DIR="openjazz-$COMMIT"
+PATCHES="disable_network.patch"
+
+ARCHITECTURES="?x86 ?x86_gcc2 x86_64 ?ppc ?arm"
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Pls reorder: gcc2, x86, x86_64, others

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+PROVIDES="
+	cmd:openjazz = $portVersion
+	"
+REQUIRES="
+        haiku
+		libsdl
+		libmodplug
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Remove one tab

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		zlib
+        "
+
+BUILD_REQUIRES="
+        haiku_devel
+		libsdl_devel
+		libmodplug_devel
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Remove one tab

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+		zlib_devel
+		pkgconfig
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Pkgconfig should be cmd:pkg_config
It needs to be in build_prereq

File: games-action/openjazz/patches/disable_network.patch [hunk 8]
```diff
@@ -0,0 +1,15 @@
+diff --git a/Makefile b/Makefile
+index 138a541..7f9cbac 100644
+--- a/Makefile
++++ b/Makefile
+@@ -7,8 +7,8 @@ CXXFLAGS ?= -g -Wall -O2
+ 
+ CXXFLAGS += -DSCALE
+ 
+-# Network support
+-CXXFLAGS += -DUSE_SOCKETS
```
Comment (extrowerk) [inline games-action/openjazz/patches/disable_network.patch]: Umm.
Haiku got socket, this Patch not needed.

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        "
+BUILD_PREREQUIRES="
+		cmd:gcc
+		cmd:make
+		"
+
+BUILD()
+{
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Try to add:
export LDFLAGS="-lnetwork"

If still not ok, try to add this TOO:
export CFLAGS="-D_BSD_SOURCE"

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+"
+HOMEPAGE="https://github.com/AlisterT/openjazz"
+COPYRIGHT="Alister Thomson (alister_j_t at yahoo dot com)"
+LICENSE="GNU GPL v2"
+REVISION="1"
+COMMIT="0f59db3a6b0e2c076b39917dc65cfb3562611ee6"
+SOURCE_URI="https://github.com/AlisterT/openjazz/archive/$COMMIT.tar.gz"
+CHECKSUM_SHA256="55c82328ef3f0185951bd339be55411410d1fa786f391091e15b7a093a08fafd"
+SOURCE_DIR="openjazz-$COMMIT"
+PATCHES="disable_network.patch"
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: Shouldn't be required.

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+BUILD_REQUIRES="
+        haiku_devel
+		libsdl_devel
+		libmodplug_devel
+		zlib_devel
+		pkgconfig
+        "
+BUILD_PREREQUIRES="
+		cmd:gcc
+		cmd:make
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: add cmd:pkg_config

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+		"
+
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	mkdir -p $binDir
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: It should go in $appsDir as it got GUI

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+	cp OpenJazz $binDir/
+}
```
Comment (extrowerk) [inline games-action/openjazz/openjazz-2017.12.02.recipe]: A nice icon and a deskbar entry would be great....

**Comment by przemub** [comment]
Thanks for the review. I have added the requested changes. If someone is able to create a better icon, then please feel free to do so :)

**Comment by przemub** [comment]
Thanks for the review. I have added the requested changes. If someone is able to create a better icon, then please feel free to do so :)

**Comment by przemub** [comment]
Thanks for the review. I have added the requested changes. If someone is able to create a better icon, then please feel free to do so :)

**Comment by przemub** [comment]
Please wait a second, it doesn't work as supposed yet...

**Comment by przemub** [comment]
It should be ok now :)

File: games-action/openjazz/openjazz-2017.12.02.recipe [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+	libsdl
+	libmodplug
+	zlib
+    "
+
+BUILD_REQUIRES="
+    haiku_devel
+	libsdl_devel
+	libmodplug_devel
+	zlib_devel
```
[the remark this answers is not in the document]
Comment (Begasus) [reply games-action/openjazz/openjazz-2017.12.02.recipe]: iirc it should be devel:libz ...

[the remark this answers is not in the document]
Comment (extrowerk) [reply games-action/openjazz/openjazz-2017.12.02.recipe]: True

**Comment by przemub** [comment]
Hi, thanks for the review. I'm going to commit changes to the upstream (finding data dir and showing an error message when there is no data) and then continue from there.

**Comment by carstene1ns** [comment]
Hoi, just getting back here after half a year. I merged the patch upstream, but this involved yet another manually maintained `Makefile`. As it is possible to use autotools for Haiku, I am going to add the needed changes there instead. As @przemub seems to be MIA currently, maybe we can move this to another pull request or is force-pushing ok here?
I assume we would only need to do the usual autotools dance in the `BUILD()` function and change the url to a tagged release. Anyway, open to suggestions, as I would really like to have OJ on more platforms.

**Comment by Begasus** [comment]
Thanks for your work on this one @przemub ! I've bumped into the new version (after your patch has been upstreamed), closing this one as done :)

**Pull request closed without merging**